lib/asset/baseAsset.js

```diff
@@ -69,6 +72,51 @@ class BaseAsset {
         return this._asset_id;
     }
 
+    /**
+     * Set multiple metadata at once.
+     * @param {Object} metadata - The metadata can be:
+     * @param {string} metadata.title - See {@link BaseAsset#set_title}.
+     * @param {string} metadata.synopsis - See {@link BaseAsset#set_synopsis}.
+     * @param {ImageAsset} metadata.thumbnail - See {@link BaseAsset#set_thumbnail}.
+     * @param {string} metadata.canonical_uri - See {@link BaseAsset#set_canonical_uri}.
+     * @param {(Date|number|string)} metadata.last_modified_date - See
+     * {@link BaseAsset#set_last_modified_date}.
+     * @param {(Date|number|string)} metadata.date_published - See
+     * {@link BaseAsset#set_date_published}.
+     * @param {number} metadata.sequence_number - See {@link BaseAsset#set_sequence_number}.
+     * @param {string} metadata.license - See {@link BaseAsset#set_license}.
+     * @param {string[]} metadata.tags - See {@link BaseAsset#set_tags}.
+     */
+    set_metadata(metadata) {
+        if (typeof metadata.title !== 'undefined') {
+            this.set_title(metadata.title);
+        }
+        if (typeof metadata.synopsis !== 'undefined') {
+            this.set_synopsis(metadata.synopsis);
+        }
+        if (typeof metadata.thumbnail !== 'undefined') {
+            this.set_thumbnail(metadata.thumbnail);
+        }
+        if (typeof metadata.canonical_uri !== 'undefined') {
+            this.set_canonical_uri(metadata.canonical_uri);
+        }
+        if (typeof metadata.last_modified_date !== 'undefined') {
+            this.set_last_modified_date(metadata.last_modified_date);
+        }
+        if (typeof metadata.date_published !== 'undefined') {
+            this.set_date_published(metadata.date_published);
+        }
+        if (typeof metadata.sequence_number !== 'undefined') {
+            this.set_sequence_number(metadata.sequence_number);
+        }
+        if (typeof metadata.license !== 'undefined') {
+            this.set_license(metadata.license);
+        }
+        if (typeof metadata.tags !== 'undefined') {
+            this.set_tags(metadata.tags);
+        }
```

Comment:
I think it would be nice to print out warnings, or even error out, for any metadata key that was passed-through but ends up not being used.

Reply:
+1